### PR TITLE
Feature solver

### PR DESCRIPTION
**Pull request by siddghosh108**
In this PR:

- We created a class called solver
- We created a method called factorial and implemented it
- We created a method called reverse and implemented it
- We created a method called fizzbuzz and implemented it

Please give your review

### CODE REVIEW

File: spec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      expect(solver.fizzbuzz(3)).to eq 'fizz'
+    end
+    it 'returns buzz when given 5' do
+      solver = Solver.new
+      expect(solver.fizzbuzz(5)).to eq 'buzz'
+    end
+    it 'returns 1 when given 1' do
+      solver = Solver.new
+      expect(solver.fizzbuzz(1)).to eq '1'
+    end
```
Comment (V-Blaze) [inline spec/solver_spec.rb]: - You have done great so far :+1: But there are a few changes you need to make before moving to the next milestone. :smile: :+1: 

- - Screenshot reference:
Selection_036

- [ ] You did great implementing the tests for the `fizzbuzz`.  I've taken a close look at the test cases for the `fizzbuzz` method, and I appreciate the effort you've put into it. :+1: :100: 
However, it seems there's a slight misinterpretation in the expected behavior. The requirement for the `fizzbuzz` method is to return "fizz" when the given number is divisible by 3, and  `'buzz' when N is divisible by 5 .. etc. Currently, the test case for divisibility by 3 is checking against the number 3 itself.

To align with the specification, please update the test case to check for a number that is divisible by 3. Here's an example:

```ruby
it 'returns fizz when given a number divisible by 3' do
  solver = Solver.new
  expect(solver.fizzbuzz(6)).to eq 'fizz'
end
```
You need to also update the `it bock` of the other test cases to better articulate the requirement i.e:  `returns "buzz" When N is divisible by 5`, `returns "fizzbuzz" When N is divisible by 5`, `returns N as a string when N is not divisible by 3 and 5`

File: spec/solver_spec.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
+require_relative '../solver'
+
+describe Solver do
```
Comment (V-Blaze) [inline spec/solver_spec.rb:3]: - - Requirement screenshot:
Screenshot from 2023-11-16 09-29-30

- [ ] Observing your commit history, it appears that the `Test-Driven Development (TDD)` approach might not have been strictly followed. The typical TDD cycle involves writing tests first, `committing, pushing`, and then proceeding with the implementation of the class and method. This practice ensures a clear roadmap for development and enhances the overall robustness of the codebase. Consider adhering to this cycle for future contributions to reinforce code quality and adherence to specifications. :smile: :small_airplane:

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @siddghosh108, @Goldinium  

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

giphy


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- Class implemented correctly ✔
-  All class methods are present👋 
-  Current implemented tests are passing✔

 ## Required Changes ♻️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/solver_spec.rb (hunk 2 above)
Comment (siddghosh108) [reply spec/solver_spec.rb:3]: I do agree and will follow your sugesstions in future

File: spec/solver_spec.rb (hunk 1 above)
Comment (siddghosh108) [reply spec/solver_spec.rb]: I have made the exact changes you asked for, please see the below code:
describe '#fizzbuzz' do
    solver = Solver.new
    it 'returns fizzbuzz when given a number divisible by both 3 and 5' do
      expect(solver.fizzbuzz(15)).to eq 'fizzbuzz'
    end
    it 'returns fizz when given a number divisible by 3' do
      expect(solver.fizzbuzz(6)).to eq 'fizz'
    end
    it 'returns buzz when given a number divisible by 5' do
      expect(solver.fizzbuzz(5)).to eq 'buzz'
    end
    it 'returns the number as a string when given a number not divisible by 3 and 5' do
      expect(solver.fizzbuzz(1)).to eq '1'
    end
  end

**Review by TadesseDev — APPROVED** [review]
### Hi @siddghosh108 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
AmazingCreepyFrogmouth-max-1mb




## Highlights
 - [x] Good job making the changes requested by the previous code reviewer. ✅


## Optional suggestions

- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**